File: 7_binary_search/이준우/징검다리.kt
class Solution {
    fun solution(distance: Int, rocks: IntArray, n: Int): Int {
        var answer = 0
        val _rocks = rocks.toMutableList()
        _rocks.add(distance)
        _rocks.sort()

        var left = 1
        var right = distance

        while (left <= right) {
            val minGap = (left + right) / 2

            var removed = 0
            var current = 0

            for (rock in _rocks) {
                if (rock - current < minGap) removed++
                else current = rock
            }

            if (removed > n) {
                right = minGap - 1
            } else {
                answer = minGap
                left = minGap + 1
            }
        }

        return answer
    }
}

File: 7_binary_search/이준우/징검다리.py
def solution(distance, rocks, n):
    answer = 0

    rocks.append(distance)
    rocks.sort()
    left = 0
    right = distance

    while left <= right:
        min_gap = (left + right) // 2

        current = removed = 0
        for rock in rocks:
            if rock - current < min_gap:
                removed += 1
            else:
                current = rock

        if removed > n:
            right = min_gap - 1
        else:
            left = min_gap + 1
            answer = min_gap

    return answer
